app/views/static/payment.html.erb

```diff
@@ -0,0 +1,92 @@
+<html>
+<meta name="viewport" content="width=device-width, initial-scale=1" />
+<head>
+<%= csrf_meta_tags %>
+<script src="https://js.stripe.com/v3/"></script>
+<script>
+  function stripeTokenHandler(token) {
+    // Insert the token ID into the form so it gets submitted to the server
+    var form = document.getElementById('payment-form');
+    var hiddenInput = document.createElement('input');
+    hiddenInput.setAttribute('type', 'hidden');
+    hiddenInput.setAttribute('name', 'stripeToken');
+    hiddenInput.setAttribute('value', token.id);
+    form.appendChild(hiddenInput);
+
+    // Submit the form
+    fetch('/charge', {
+      method: 'POST',
+      body: JSON.stringify({token: token.id}),
+      credentials: 'same-origin',
+      headers: {
+        'content-type': 'application/json',
+        'X-CSRF-Token': $('meta[name="csrf-token"]').attr('content')
+      }
+    }).then((response) => {
+      if (response.status  === 400) {
+        alert('Bad Request -- show user a pretty error message');
+        // You can fail charge in development with the card number that
+        // fails cvc_check here: https://stripe.com/docs/testing#cards
+      }
+    });
+  }
+
+  function tokenizeCard(card) {
+    // Create a token or display an error when the form is submitted.
+    var form = document.getElementById('payment-form');
+    form.addEventListener('submit', function(event) {
+      event.preventDefault();
+
+      stripe.createToken(card).then(function(result) {
+        if (result.error) {
+          // Inform the customer that there was an error.
+          var errorElement = document.getElementById('card-errors');
+          errorElement.textContent = result.error.message;
+        } else {
+          // Send the token to your server.
+          stripeTokenHandler(result.token);
+        }
+      });
+    });
+  }
+
+  var stripe = Stripe("<%=ENV['STRIPE_PUBLISHABLE_KEY']%>");
```

Comment:
Getting keys is a little tricky. Add to the application layout (layouts/application.html.erb) this code:

```
window.STRIPE_PUBLISHABLE_KEY = <%= ENV[...] %>
```

and then update this line to use the `window` variable instead